mmpose/models/mesh_heads/discriminator.py

```diff
@@ -0,0 +1,270 @@
+                print('use_ac_func require list, '
+                      'get {}'.format(type(self.use_ac_func)))
+                break
+
+            l_fc_layer = len(self.fc_layers)
+            l_use_drop = len(self.use_dropout)
+            l_drop_porb = len(self.drop_prob)
+            l_use_ac_func = len(self.use_ac_func)
+
+            return l_fc_layer >= 2 and \
+                l_use_drop < l_fc_layer and \
+                l_drop_porb < l_fc_layer and \
+                l_use_ac_func < l_fc_layer and \
+                l_drop_porb == l_use_drop
+
+        return False
+
+    def create_layers(self):
+        """Create layers."""
+        l_fc_layer = len(self.fc_layers)
+        l_use_drop = len(self.use_dropout)
+        # l_drop_porb = len(self.drop_prob)
+        l_use_ac_func = len(self.use_ac_func)
+
+        self.fc_blocks = nn.Sequential()
+
+        for _ in range(l_fc_layer - 1):
+            self.fc_blocks.add_module(
+                name='regressor_fc_{}'.format(_),
+                module=nn.Linear(
+                    in_features=self.fc_layers[_],
+                    out_features=self.fc_layers[_ + 1]))
+
+            if _ < l_use_ac_func and self.use_ac_func[_]:
+                self.fc_blocks.add_module(
+                    name='regressor_af_{}'.format(_), module=nn.ReLU())
+
+            if _ < l_use_drop and self.use_dropout[_]:
+                self.fc_blocks.add_module(
+                    name='regressor_fc_dropout_{}'.format(_),
+                    module=nn.Dropout(p=self.drop_prob[_]))
+
+    def forward(self, inputs):
+        """Forward function."""
+        msg = 'the base class [LinearModel] is not callable!'
+        sys.exit(msg)
+
+
+class ShapeDiscriminator(LinearModel):
+    """
+    Discriminator for SMPL shape parameters, the inputs is (batch size x 10)
+    Args:
+        fc_layers (List): List of neuron count,
+         such as [2133, 1024, 1024, 85]
+        use_dropout (List): List of bool define use dropout or
+            not for each layer, such as [True, True, False]
+        drop_prob (List): List of float defined the drop prob,
+            such as [0.5, 0.5, 0]
+        use_ac_func(List): List of bool define use active
+            function or not, such as [True, True, False]
+    """
+
+    def __init__(self, fc_layers, use_dropout, drop_prob, use_ac_func):
+        if fc_layers[-1] != 1:
+            msg = 'the neuron count of the last layer' \
+                  ' must be 1, but got {}'.format(fc_layers[-1])
+            sys.exit(msg)
+
+        super(ShapeDiscriminator, self).__init__(fc_layers, use_dropout,
+                                                 drop_prob, use_ac_func)
+
+    def forward(self, inputs):
+        """Forward function."""
+        return self.fc_blocks(inputs)
+
+
+class PoseDiscriminator(nn.Module):
+    """Discriminator for SMPL pose parameters of each joints. It is composed of
+    23 discriminators for 23 joints. The inputs is (batch size x 23 x 9)
```

Comment:
Shall we make this `23` as a variable (default = 23)?

Reply:
OK, I will modify the codes.